codex/sales/states/unknown.nim

```diff
@@ -51,3 +52,5 @@ method run*(state: SaleUnknown, machine: Machine): Future[?State] {.async.} =
     return some State(SaleFinished())
   of SlotState.Failed:
     return some State(SaleFailed())
+  of SlotState.Cancelled:
+    return some State(SalePayout())
```

Comment:
Why not go to the Cancelled state instead?

Reply:
Yup you are right, that is a left-over from an earlier version where I used the `SalePayout` state for freeing the slot, but then thought it will be cleaner to have the `freeSlot` calls directly in the states... Will fix that.